packages/core-mobile/app/contexts/DeeplinkContext/DeeplinkContext.tsx

```diff
@@ -39,78 +41,96 @@ export const DeeplinkContextProvider = ({
   children: React.ReactNode
 }): JSX.Element => {
   const dispatch = useDispatch()
+  const store = useStore()
   const walletState = useSelector(selectWalletState)
   const isWalletActive = walletState === WalletState.ACTIVE
   const isDeveloperMode = useSelector(selectIsDeveloperMode)
   const isNotificationBlocked = useSelector(selectIsNotificationBlocked)
   const processedFeatureFlags = useSelector(selectFeatureFlags)
   const [pendingDeepLink, setPendingDeepLink] = useState<DeepLink>()
+  const { getNetwork } = useNetworks()
+
+  const maybeSetActiveNetwork = useCallback(
+    (data: NotificationData) => {
+      if (
+        'chainId' in data &&
+        ['string', 'number'].includes(typeof data.chainId)
+      ) {
+        const chainId = Number(data.chainId)
+        const network = getNetwork(chainId)
+        //check if testnet should be toggled to match chainId provided in data
+        if (network && network.isTestnet !== isDeveloperMode) {
+          dispatch(toggleDeveloperMode())
+        }
+        dispatch(setActive(chainId))
+      }
+    },
+    [dispatch, getNetwork, isDeveloperMode]
+  )
 
-  const expireDeepLink = useCallback(() => {
-    setPendingDeepLink(undefined)
-  }, [])
+  const maybeSetActiveAccount = useCallback(
+    (data: NotificationData) => {
+      if ('accountAddress' in data && typeof data.accountAddress === 'string') {
+        const account = selectAccountByAddress(data.accountAddress)(
+          store.getState() as RootState
+        )
```

Comment:
let's try not to use `store` here as well. you can get the list of accounts outside and do a simple lookup here for it. you can create a hook for it if you want to be fancy.

Reply:
Will be updated in following PR

Reply:
[Move setting active network and account to listeners](https://github.com/ava-labs/core-mobile/pull/1659/commits/c627dd6853b9ec197ab3592568f708a1953d07bc)